algorithms/Python/linked_lists/middle-node-linkedlist.py

```diff
@@ -0,0 +1,41 @@
+#printing middle node element of linked list
+#we are using two pointer method to get the solution
+
+class Node:
+    #Constructor to build node
+    def __init__(self, data):
+        self.data = data
+        self.next = None
+
+class LinkedList:
+    #Constructor to create head node
+    def __init__(self):
+        self.head = None
+
+    #Function to push element at head (at beginning)
+    def push(self, data):
+        new_node = Node(data)
+        new_node.next = self.head
+        self.head = new_node
+    
+    #Function returns data of middle node
+    def middle_element(self):
+        slow = self.head
+        fast = self.head
+        while fast and fast.next:
+            slow = slow.next
+            fast = fast.next.next
+        return slow.data
```

Comment:
what if the list is empty ?

Reply:
So if the list is empty we can add two lines of code in starting of the middle_element function :-
if self.head is None:
    return None

Reply:
indeed, that should do the trick.

Reply:
I have updated the code for empty list. Please check it and let me know

Reply:
Thanks you